### ISSUE TITLE
503 reported on Discovery without targets defined

### ISSUE THREAD

**Issue by eaglerainbow** [promregator/promregator#180]
## Environment
branch `master`, 0.8.0 prior to delivery

## Summary / Problem Statement
If there are no targets defined in the configuration file, hitting the `/discovery` results in a 503 response. The response has changed (regression issue) - 0.7.x reports 200 with body `[]`

## Observed Behavior
Reporting 503

## Expected Behavior
HTTP status code 200 with a body of 
```json
[]
```

**Comment by eaglerainbow** [comment]
It seems to got introduced with e04d34d18f09635c0df5341ee845ad6716e42060.
The commit's description is SonarQube: Simplify List handling
``` 
-        if (instances == null) {
+        if (instances == null || instances.isEmpty()) {
```

**Comment by eaglerainbow** [comment]
Fixed with ecf9a00d34de41b1094e4a846d431c31bb654c65
